public/javascripts/controllers.js

```diff
@@ -1,692 +1,816 @@
 angular.module('uforia')
 
-.controller('appCtrl', function($rootScope){
-  $rootScope.Utils = {
-     keys : Object.keys
-  }
-})
-
-.controller('searchCtrl', function($rootScope, $scope, $http, $modal, $timeout, types) {
-  $scope.queryMatchesCount = 0;
-  $scope.maxResultCount = 0;
-
-  $scope.searchTypes = types;
-  $scope.searchType = $scope.searchTypes[0];
-  $scope.parameters = [{operator: "must", andOr: "And"}];
-  var visualization = {};
-
-  $scope.$watch('viewType', function(newVal, oldVal){
-    if(newVal){
-      removeSVG();
-      changeScripts(function(){
-        if($scope.searchForm.$valid && $scope.queryMatchesCount < $scope.maxResultCount){
-          $scope.search();
+    .controller('appCtrl', function($rootScope){
```

Comment:
I don't think you need an app controller. I looks like you're only trying to set a global variable but the right place for that is the config of the app. Check out the logout function and the isLoggedIn global vars. That should give you and idea of how those things are done.